src/sinks/loki/sink.rs

```diff
@@ -158,19 +160,30 @@ pub(super) struct EventEncoder {
 
 impl EventEncoder {
     fn build_labels(&self, event: &Event) -> Vec<(String, String)> {
-        self.labels
-            .iter()
-            .filter_map(|(key_template, value_template)| {
-                if let (Ok(key), Ok(value)) = (
-                    key_template.render_string(event),
-                    value_template.render_string(event),
-                ) {
-                    Some((key, value))
+        let mut vec: Vec<(String, String)> = Vec::new();
+
+        for (key_template, value_template) in self.labels.iter() {
+            if let (Ok(key), Ok(value)) = (
+                key_template.render_string(event),
+                value_template.render_string(event),
+            ) {
+                if let Some(opening_prefix) = key.strip_suffix('*') {
+                    let output: serde_json::map::Map<String, serde_json::Value> =
```

Comment:
It is a minor point, but `unwrap_or_default` will create an unnecessary allocation in the error path which could be avoided by something along the lines of:

```suggestion
                    let output: Result<serde_json::map::Map<String, serde_json::Value>, _> =
                        serde_json::from_str(value.as_str());
                        
                    if let Ok(output) = output {
```